### PR TITLE
feat(developer): add web font references to package 🎺

### PR DESCRIPTION
**Pull request by mcdurdin**
While a package supports a .ttf or .otf font for the OSK and Display fonts for use within the Keyman apps, on websites it also supports .woff and .woff2. This change makes it possible to specify a set of fonts for a package which will be provided through api.keyman.com to websites.

This means that .woff and .woff2 fonts may also be included in the package and may not be used on target devices at this time, it further opens the pathway to deploying .kmp packages to web in the future.

The deployment stage of keymanapp/keyboards will require a new phase to copy web fonts from packages to s.keyman.com.

This needs corresponding PRs against api.keyman.com, help.keyman.com, and keyboards repositories:

- [ ] keymanapp/api.keyman.com (schema updates)
- [ ] keymanapp/help.keyman.com (documentation updates)
- [ ] keymanapp/keyboards (s.keyman.com font copy phase, keymanapp/keyboards#2323)
- [ ] keymanapp/keyboards (revised keyboards, blocked until kmc is applied on the repo)

# User Testing

All these tests should be performed in the Package Editor in Keyman Developer. To begin, create a package a keyboard and multiple fonts, preferably including .ttf, .otf, .woff, and .woff2 fonts. It does not matter for the purpose of this test what language the fonts support, just that there are multiple different types.

## GROUP_SET_WEB_FONT

1. In the Keyboards tab, select a keyboard, and click "Web Fonts" button.
2. The check list box should list all the fonts that you added in the Files tab.
3. Select 1 or more fonts from the list and click OK.
4. The fonts should be listed in the label to the right of the button.
5. Save the package, close and reopen it.
6. The fonts should still be listed in the label to the right of the button.
7. Click the "web fonts" button again; the fonts should be checked in the list.
8. Uncheck one or more fonts, and click OK. The newly checked fonts should be listed in the label to the right of the button.

* TEST_SET_WEB_KEYBOARD_FONT: Test this with the Web Fonts button to the right of the "Keyboard Font" combo box.
* TEST_SET_WEB_DISPLAY_FONT: Test this with the Web Fonts button to the right of the "Display Font" combo box.

### CODE REVIEW

**Comment by bharanidharanj** [comment]
# Test Results

### GROUP_SET_WEB_FONT: 

* **TEST_SET_WEB_KEYBOARD_FONT (PASSED):** Tested with the attached PR build (Keyman Developer 17.0.159-alpha-test-9463) in Windows 11 OS (VM) and here is my observation : 1. Added multiple fonts in a newly created package. 2. Able to see the added fonts appeared to the right side of the Keyboard font / Web fonts button. 3. Able to change the fonts and save it using the Save button. Verified that the newly added fonts be listed in the label to the right of the button. Seems to be working as expected.

   



* **TEST_SET_WEB_DISPLAY_FONT (PASSED):** 1. Added multiple fonts in a newly created package. 2. Able to see the added fonts appeared to the right side of the Display font / Web fonts button. 3. Able to change the fonts and save it using the Save button. Verified that the newly added fonts be listed in the label to the right of the button.

File: common/schemas/kps/kps.xsd [hunk 1]
```diff
@@ -125,6 +125,8 @@
                   <xs:element minOccurs="0" maxOccurs="1" name="DisplayFont" type="xs:string" />
```
Comment (darcywong00) [inline common/schemas/kps/kps.xsd:125]: Would OSKFont and DisplayFont also become `km-keyboard-fonts` type?

Comment (darcywong00) [reply common/schemas/kps/kps.xsd:125]: Ah, I see its string vs string[]

Comment (mcdurdin) [reply common/schemas/kps/kps.xsd:125]: Yeah, not changing existing fields either.

Comment (mcdurdin) [reply common/schemas/kps/kps.xsd:125]: Which is why we have separate fields for the web fonts

**Review by darcywong00 — APPROVED** [review]
lgtm

**Review by rc-swag — APPROVED** [review]
lgtm

**Pull request merged**